### DISCUSSION TITLE
Websocket과 STOMP

### DISCUSSION THREAD

**Discussion by amaran-th** [BDD-CLUB/challenge-every-week-posting#543] [1. 개발 관련 글 모음]
이번에 운좋게 websocket을 다뤄볼 기회가 생겨서 글로 정리해보았습니다.

[Websocket과 STOMP](https://amaran-th.github.io/JavaScript/Websocket%EA%B3%BC%20STOMP/)
[websocket으로 서버와 실시간 통신 구현하기](https://amaran-th.github.io/React/Websocket%EC%9C%BC%EB%A1%9C%20%EC%84%9C%EB%B2%84%EC%99%80%EC%9D%98%20%EC%8B%A4%EC%8B%9C%EA%B0%84%20%ED%86%B5%EC%8B%A0%20%EA%B5%AC%ED%98%84%ED%95%98%EA%B8%B0/)

**Comment by jojaeng2** [comment]
저도 websocket + stomp로 프로젝트 진행했던적이 있어서 재밌게 읽었습니다.

읽다보니 궁금한점이 생겼는데요. 아키텍처와 요구사항을 봤을때 고객의 요청이 들어오면 사장님에게 알려주는 구조 같은데 (고객 -> 사장님)
사장님 -> 고객 방향의 전달이 없다면 websocket이 아닌 SSE(Server Send Event)가 서버의 관점에서 리소스를 더 효율적으로 쓸 수 있을 것 같다는 생각이 들었는데 (connection을 맺고 있는 스레드가 사장님과 연결만 되면 되므로) 굳이 websocket을 선택하신 이유가 있으신가요?

**Reply by shkisme** [reply]
🤔 이건 저도 동일하게 의문을 품었던 부분이었는데요 ! 기술 결정을 하신 분 말씀에 따르면, 

> Websocket이 확장성이 더 좋을 것 같아서 골랐다. 기술숙련도 차이도 있고, SSE보다 웹소켓이 조금 더 유스케이스가 많아보여서 선택하게 되었다.

아마도 사장님 -> 고객 방향의 전달을 실시간으로 해야하는 상황도 가정 & 열어두기 위해서 선택한 것으로 보입니다 !
